experiment/build/generate_makefile.py

```diff
@@ -13,50 +13,193 @@
 # limitations under the License.
 """Generates Makefile containing docker image targets."""
 
+from experiment.build import docker_images
+from common import yaml_utils
 import argparse
+import os
 
-from experiment.build import docker_images
+BASE_TAG = "gcr.io/fuzzbench"
+BENCHMARKS_DIR = os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, 'benchmarks')
+FUZZERS_DIR = os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, 'fuzzers')
 
+RUN_TARGET_TEMPLATE = """
```

Comment:
OK, this is blowing up :) Consider having a *single* template per docker *image* (object returned by get_buildable_images()). Most images would only have one `.build-{image_name}` rule, however the final fuzzer-benchmark runner images would have run/test/debug rules as well. It would look something like this in jinja:
```
.build-{{image.name}}: ...
   ...

{% if image_name.endswith('-runner') %}

run-{{image.fuzzer}}-{{image.benchmark}}: .build-{{image.name}}:
  ...

debug-{{image.fuzzer}}-{{image.benchmark}}: .build-{{image.name}}:
  ...

{% endif %}

```

Reply:
Added template item for run image.

Reply:
I see you added a `'{{run_template}}-{fuzzer}-{benchmark}'` to image_types.yaml. I was referring to the makefile rule template here in this file where this comment is on. The image_types.yaml describes image types and their dependencies. A makefile rule like `run-fuzzerX-bencmarkY` or `debug-fuzzerX-benchmarkY` is not an image type. The jinja template sketch above is for generating the Makefile rules using the objects returned by the get_images_to_build() function.

Reply:
Added as template. Not as Jinja template explicitly, python string looked more readable. But could change if required.